src/Frame.jsx

```diff
@@ -45,6 +45,7 @@ export class Frame extends Component {
 
     const doc = this.getDoc();
     if (doc && doc.readyState === 'complete') {
+      this.handleLoad();
```

Comment:
So this seems to break for everyone outside of next.js so I'm wondering what next.js is doing to have to trigger the load event manually?

Reply:
This `readyState` code is here to handle cross browser differences

- Chromium based browsers don't trigger onload and need the `forceUpdate` to render and trigger the load event
- Firefox doesn't use `readyState` and instead will update contents on iframe load

If we introduce a third call then chromium based browsers throw as it tries to render the contents before srcdoc has injected it (srcdoc is async) so there may be a timing issue here. I'm unsure why this works in next.js based apps in chromium based browsers??

Firefox works fine with as this addition is not a codepath firefox reaches

Reply:
I could definitely see this being an async timing issue. My `initialContent` is fairly large and includes external stylesheets like FontAwesome and stuff, which I could see causing some timing issue if it takes too long to load, especially on a page refresh where external resources might have to reload but other things might be cached and thus trigger things in a different order, but randomly on a later refresh the timing is off in the other direction and it works fine.

Reply:
> My initialContent is fairly large and includes external stylesheets like FontAwesome and stuff, which I could see causing some timing issue if it takes too long to load

Interesting, do you think you could share a test case with your intitialContent?

In some further exploring it seems the `addEventListener` in `compnentDidMount` is superfluous and actually triggers a double load event in firefox.

Reply:
I would think you could just take @baptisteArno 's [test case](https://github.com/baptisteArno/next-react-frame-bug) and apply my initialContent over the top of it.

```
initialContent : `<!DOCTYPE html><html><head>
	<link href="//use.fontawesome.com/releases/v5.15.1/css/all.css" rel="stylesheet" />
	<link href="//fonts.googleapis.com/css?family=Open+Sans:400,300,600,700" rel="stylesheet" type="text/css" />
	<base target=_blank>
	</head><body style='overflow: hidden'><div></div></body></html>`
```

You can also try using my repo https://github.com/naturalcrit/homebrewery and just update react-frame-component to see it break. Unfortunately I'm having trouble isolating a smaller version of the app because it's inherited from someone else and I don't know all the ins and outs.

Reply:
> So this seems to break for everyone outside of next.js so I'm wondering what next.js is doing to have to trigger the load event manually?

Did you mean this PR breaks for everyone not using next.js or the original issue occurs for everyone not using next.js?

Reply:
Sorry to bombard this here. I may have stumbled onto something else:

I suspect the following is happening, at least for me:

1) User has already visited the page recently and has it cached in their browser.
2) User visits the page again, or navigates back to it using the browser history.
3) `componentDidMount` fires, and everything is cached, so `doc.readyState == complete`.  The `else` statement is never reached and the `load` event listener is never added.

Reply:
@calculuschild the iframe itself has a `onLoad` prop that should fire the else was only ever for firefox

Reply:
> Did you mean this PR breaks for everyone not using next.js or the original issue occurs for everyone not using next.js? I can verify this PR fixes the issue for me. I'm using Chrome on node.js, not next.js.

This PR breaks, see the github action tests. It complains about the `mountTarget` being null as the `inititalContent` provided to the iframe via `srcdoc` hasn't rendered yet.

It might be that next.js has aggressive caching and that's why it manifests itself there more than elsewhere

Reply:
The `onLoad` property on the iframe.... should it be `onload`, lowercase? https://www.w3schools.com/jsref/event_onload.asp

Reply:
it's a react thing it'll get normalised to `onload` when it's written to the DOM

Reply:
Gotcha. Like className to class. Ok.

Another possibility. Seems like if a page is in cache, onLoad can potentially trigger right away, before the handler is even assigned to onLoad.
https://stackoverflow.com/questions/19917436/window-onload-not-running-on-page-refresh

Reply:
thanks seems like there could be something related there. This is issue is hard to diagnose as it hasn't been easy for me to reproduce if someone could create a codesandbox.io where it is easily recreated it would help a lot.